gstreamer/src/toc.rs

```diff
@@ -0,0 +1,350 @@
+use std::ffi::CStr;
+use std::mem;
+
+use ffi;
+
+use glib;
+use glib::translate::{from_glib, from_glib_full,
+                      FromGlibPtrContainer, ToGlib, ToGlibPtr};
+
+use miniobject::*;
+use TocEntryType;
+use TocScope;
+use TocLoopType;
+use TagList;
+use TagMergeMode;
+
+pub type Toc = GstRc<TocRef>;
+pub struct TocRef(ffi::GstToc);
+
+unsafe impl MiniObject for TocRef {
+    type GstType = ffi::GstToc;
+}
+
+impl GstRc<TocRef> {
+    pub fn new(scope: TocScope) -> Self {
+        assert_initialized_main_thread!();
+        unsafe { from_glib_full(ffi::gst_toc_new(scope.to_glib())) }
+    }
+}
+
+impl TocRef {
+    pub fn get_scope(&self) -> TocScope {
+        unsafe { from_glib(ffi::gst_toc_get_scope(self.as_ptr())) }
+    }
+
+    pub fn find_entry(&self, uid: &str) -> Option<TocEntry> {
+        unsafe {
+            let toc_entry = ffi::gst_toc_find_entry(
+                self.as_ptr(),
+                uid.to_glib_none().0,
+            );
+            if toc_entry.is_null() {
+                return None;
+            }
+
+            Some(TocEntry::from_glib_borrow(
```

Comment:
This is unsafe. Either return a &TocEntryRef or from_glib_none()

from_glib_borrow() return values must only be used in this very scope (and not be passed outside) and only references be passed to functions called from this scope